mopidy/audio/actor.py

```diff
@@ -232,37 +232,39 @@ def teardown_message_handling(self):
         self._message_handler_id = None
 
     def teardown_event_handling(self):
-        self._pad.remove_event_probe(self._event_handler_id)
+        self._pad.remove_probe(self._event_handler_id)
         self._event_handler_id = None
 
     def on_message(self, bus, msg):
-        if msg.type == gst.MESSAGE_STATE_CHANGED and msg.src == self._element:
+        if (
+                msg.type == Gst.MessageType.STATE_CHANGED and
+                msg.src == self._element):
             self.on_playbin_state_changed(*msg.parse_state_changed())
-        elif msg.type == gst.MESSAGE_BUFFERING:
-            self.on_buffering(msg.parse_buffering(), msg.structure)
-        elif msg.type == gst.MESSAGE_EOS:
+        elif msg.type == Gst.MessageType.BUFFERING:
+            self.on_buffering(msg.parse_buffering(), msg.get_structure())
+        elif msg.type == Gst.MessageType.EOS:
             self.on_end_of_stream()
-        elif msg.type == gst.MESSAGE_ERROR:
+        elif msg.type == Gst.MessageType.ERROR:
             self.on_error(*msg.parse_error())
-        elif msg.type == gst.MESSAGE_WARNING:
+        elif msg.type == Gst.MessageType.WARNING:
             self.on_warning(*msg.parse_warning())
-        elif msg.type == gst.MESSAGE_ASYNC_DONE:
+        elif msg.type == Gst.MessageType.ASYNC_DONE:
             self.on_async_done()
-        elif msg.type == gst.MESSAGE_TAG:
+        elif msg.type == Gst.MessageType.TAG:
             self.on_tag(msg.parse_tag())
-        elif msg.type == gst.MESSAGE_ELEMENT:
-            if gst.pbutils.is_missing_plugin_message(msg):
+        elif msg.type == Gst.MessageType.ELEMENT:
+            if GstPbutils.is_missing_plugin_message(msg):
                 self.on_missing_plugin(msg)
 
     def on_event(self, pad, event):
-        if event.type == gst.EVENT_NEWSEGMENT:
+        if event.type == Gst.EventType.SEGMENT:
             self.on_new_segment(*event.parse_new_segment())
-        elif event.type == gst.EVENT_SINK_MESSAGE:
+        elif event.type == Gst.EventType.SINK_MESSAGE:
             # Handle stream changed messages when they reach our output bin.
             # If we listen for it on the bus we get one per tee branch.
             msg = event.parse_sink_message()
-            if msg.structure.has_name('playbin2-stream-changed'):
-                self.on_stream_changed(msg.structure['uri'])
+            if msg.get_structure().has_name('playbin-stream-changed'):
```

Comment:
`Gst.MessageType.STREAM_START` could also work though it depends on if we want when the whole pipeline gets it or the tee.